### DISCUSSION TITLE
On "resource strategy" and the ReuseFactor numbers

### DISCUSSION THREAD

**Discussion by wilfredkisku** [fastmachinelearning/hls4ml#654] [Q&A]
While I have a crude understanding of Strategy when assigned to be a "Resource", synthesizes into parallel realization. The question I have is in regard to the numbers that I get for valid ReuseFactor.

1. Why is it that the reuse factor cannot be 1?
2. Why are there specific numbers for the reuse factor convolution and dense layers?
3. Why the input convolution factor number is different from the other convolution layer numbers?
4. How about the output dense layer?

```
WARNING: Invalid ReuseFactor=1 with "Resource" strategy in layer "conv1". Using ReuseFactor=3 instead. Valid ReuseFactor(s): 3,9,18,36,72,144.
WARNING: Invalid ReuseFactor=1 with "Resource" strategy in layer "conv2". Using ReuseFactor=2 instead. Valid ReuseFactor(s): 2,3,4,6,8,9,12,16,18,24,36,48,72,144,288,576,1152,2304.
WARNING: Invalid ReuseFactor=1 with "Resource" strategy in layer "conv3". Using ReuseFactor=2 instead. Valid ReuseFactor(s): 2,3,4,6,8,9,12,16,18,24,36,48,72,144,288,576,1152,2304.
WARNING: Invalid ReuseFactor=1 with "Resource" strategy in layer "conv4". Using ReuseFactor=2 instead. Valid ReuseFactor(s): 2,3,4,6,8,9,12,16,18,24,36,48,72,144,288,576,1152,2304.
WARNING: Invalid ReuseFactor=1 with "Resource" strategy in layer "conv5". Using ReuseFactor=2 instead. Valid ReuseFactor(s): 2,3,4,6,8,9,12,16,18,24,36,48,72,144,288,576,1152,2304.
WARNING: Invalid ReuseFactor=1 with "Resource" strategy in layer "conv6". Using ReuseFactor=2 instead. Valid ReuseFactor(s): 2,3,4,6,8,9,12,16,18,24,36,48,72,144,288,576,1152,2304.
WARNING: Invalid ReuseFactor=1 with "Resource" strategy in layer "dense_0_2". Using ReuseFactor=2 instead. Valid ReuseFactor(s): 2,4,8,16,32,64,128,256,512,1024,2048,4096.
WARNING: Invalid ReuseFactor=1 with "Resource" strategy in layer "dense_1_2". Using ReuseFactor=2 instead. Valid ReuseFactor(s): 2,4,8,16,32,64,128,256,512.
WARNING: Invalid ReuseFactor=1 with "Resource" strategy in layer "dense_2_2". Using ReuseFactor=2 instead. Valid ReuseFactor(s): 2,4,8,16,32,64,128,256,512,1024,2048.
WARNING: Invalid ReuseFactor=1 with "Resource" strategy in layer "output_dense". Using ReuseFactor=2 instead. Valid ReuseFactor(s): 2,4,8,16,32,64,128,320,640.
```

All of these questions are in regard to an architecture that is convolutional in nature.

**Details of the TensorFlow model:**
1. Input size (32,32,1)
2. Convolution kernel (3,3) with the filter numbers to be 16
3. Output dense layer to be (10) with a softmax.

Thanks.

**Comment by wilfredkisku** [comment]
> 

Also, Is it possible to synthesize models with resource requirements greater than 4096?
